src/util/indexrange.cpp

```diff
@@ -55,51 +55,83 @@ IndexRange reverse(IndexRange arg) {
 }
 
 IndexRange intersect(IndexRange lhs, IndexRange rhs) {
-    if (lhs.start() <= lhs.end()) {
+    if (lhs.start() < lhs.end()) {
         if (rhs.start() <= rhs.end()) {
             const SINT start = std::max(lhs.start(), rhs.start());
             const SINT end = std::min(lhs.end(), rhs.end());
             if (start <= end) {
                 return IndexRange::between(start, end);
             }
         } else {
-            DEBUG_ASSERT(!"Cannot intersect index ranges with contrary orientations");
+            DEBUG_ASSERT(!"Cannot intersect index ranges with different orientations");
+            return IndexRange();
         }
-    } else {
+    } else if (lhs.start() > lhs.end()) {
         if (rhs.start() >= rhs.end()) {
             const SINT start = std::min(lhs.start(), rhs.start());
             const SINT end = std::max(lhs.end(), rhs.end());
             if (start >= end) {
                 return IndexRange::between(start, end);
             }
         } else {
-            DEBUG_ASSERT(!"Cannot intersect index ranges with contrary orientations");
+            DEBUG_ASSERT(!"Cannot intersect index ranges with different orientations");
+            return IndexRange();
+        }
+    } else {
+        DEBUG_ASSERT(lhs.start() == lhs.end());
+        if (rhs.start() <= rhs.end()) {
+            const SINT start = std::max(lhs.start(), rhs.start());
```

Comment:
lhs is a single point. The code returns this point if it is within the range of rhs. 
I would make this explicit to have a better readable code. 
if (rhs.start() <= lhs.start() && rhs.end() >= lhs.end()) {
    return lhs;
} else {
    return IndexRange();
}